crypto/merkle/simple_map.go

```diff
@@ -27,6 +27,7 @@ func (sm *simpleMap) Set(key string, value Hasher) {
 	// The value is hashed, so you can
 	// check for equality with a cached value (say)
 	// and make a determination to fetch or not.
+	// TODO: Change this to tmhash(0x00 || value.Bytes())
```

Comment:
Did we discuss to only prefix leaf nodes? According to the RFC in question, we would also prefix non-leaf nodes (with `0x01`): https://tools.ietf.org/html/rfc6962#section-2.1